### PR TITLE
[Android] Fix TextInput Cursor jumping to the right when the placeholder null

### PR DESCRIPTION
**Pull request by fabriziobertoglio1987**
## Summary

This issue fixes https://github.com/facebook/react-native/issues/28794 fixes https://github.com/facebook/react-native/issues/27658 
Flow type ?Stringish allows to set the placeholder value to null. The null value causes the cursor to jump to the right in a TextInput. The fix replaces the placeholder null value with an empty string which avoid calling setHint(null) as causes the placeholder to jump to the right. 

## Changelog

[Android] [Fixed] - avoid calling setHint with a null parameter causing cursor to jump to the right

## Test Plan

**<details><summary>CLICK TO OPEN TESTS RESULTS (28 May 2020 20a9473aaa330ad9b6e7a0b42ebd9c4ed41ce60b)</summary>**
<p>

More videos and information included in issue https://github.com/facebook/react-native/issues/28794
The below test cases are from the [following repository](https://github.com/fabriziobertoglio1987/AwesomeProject)

| **BEFORE** | 
|:-------------------------:|
| |

| **AFTER** | 
|:-------------------------:|
| |

Extensive testing on `RNTester` did not identify any regression.

| **AFTER** | 
|:-------------------------:|
| |

</p>
</details>

**<details><summary>CLICK TO OPEN TESTS RESULTS (15 February 2021 https://github.com/facebook/react-native/pull/28995/commits/20a9473aaa330ad9b6e7a0b42ebd9c4ed41ce60b)</summary>**
<p>

| **BEFORE** | 
|:-------------------------:|
| <video src="https://user-images.githubusercontent.com/24992535/107960803-5d44a980-6fa5-11eb-90e2-f0d665e35291.mp4" />|

| **AFTER** | 
|:-------------------------:|
| <video src="https://user-images.githubusercontent.com/24992535/107960602-1f478580-6fa5-11eb-8f39-b985fafa6a6c.mp4" />|

</p>
</details>

### CODE REVIEW

**Comment by devinjameson** [comment]
Why has this not been merged yet?

**Comment by safaiyeh** [comment]
cc @JoshuaGross

**Comment by Elijen** [comment]
@hramos @rickhanlonii 🙏

**Comment by fabriziobertoglio1987** [comment]
@Haris-Shahid i'll test again, but it worked when I wrote this pr...

**Comment by Yandamuri** [comment]
when will this be merged?

**Comment by sosunny** [comment]
> @Haris-Shahid i'll test again, but it worked when I wrote this pr...

Did you test it works by any chance? It didn't fix the issue for me either with RN v0.63.4 :/

**Comment by fabriziobertoglio1987** [comment]
@sosunny I tested it when I wrote the pr and you can see I have a gif showing the tests before and after applying the pr. If you want to patch the package, that is not the way to test a pr for ReactAndroid. You should just follow the guide at https://github.com/facebook/react-native/wiki/Building-from-source Thanks

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java [hunk 1]
```diff
@@ -406,7 +406,8 @@ public void setAllowFontScaling(ReactEditText view, boolean allowFontScaling) {
 
   @ReactProp(name = "placeholder")
   public void setPlaceholder(ReactEditText view, @Nullable String placeholder) {
-    view.setHint(placeholder);
+    String text = placeholder == null ? "" : placeholder;
```
Comment (fabriziobertoglio1987) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java:409]: the issue here is `@Nullable String placeholder` so I will try remove `@Nullable` option

**Comment by sreuter** [comment]
@fabriziobertoglio1987 I've noticed you just updated this PR recently. Any chance of this getting merged soon. This also just seems to happen when textAlign is set to `center`, correct?

**Review by Pedro-Saraiva-MW — APPROVED** [review]

**Comment by ShivamJoker** [comment]
Wow finally after 2 years, folks at facebook merged this PR 😳🤦

**Comment by alichampion** [comment]
In which version of react-native it will work fine?

**Comment by lunaleaps** [comment]
> In which version of react-native it will work fine?

https://github.com/facebook/react-native/wiki/Release-FAQ#when-will-my-fix-make-it-into-a-release

**Pull request closed without merging**